src/ImageSharp/Formats/Jpeg/Components/Block8x8F.cs

```diff
@@ -596,5 +600,98 @@ private static void GuardBlockIndex(int idx)
+               Unsafe.As<Vector4, Vector128<float>>(ref this.V4L),
+               1);
+
+            Vector256<float> r1 = Avx.InsertVector128(
+               Unsafe.As<Vector4, Vector128<float>>(ref this.V1L).ToVector256(),
+               Unsafe.As<Vector4, Vector128<float>>(ref this.V5L),
+               1);
+
+            Vector256<float> r2 = Avx.InsertVector128(
+               Unsafe.As<Vector4, Vector128<float>>(ref this.V2L).ToVector256(),
+               Unsafe.As<Vector4, Vector128<float>>(ref this.V6L),
+               1);
+
+            Vector256<float> r3 = Avx.InsertVector128(
+               Unsafe.As<Vector4, Vector128<float>>(ref this.V3L).ToVector256(),
+               Unsafe.As<Vector4, Vector128<float>>(ref this.V7L),
+               1);
+
+            Vector256<float> r4 = Avx.InsertVector128(
+               Unsafe.As<Vector4, Vector128<float>>(ref this.V0R).ToVector256(),
+               Unsafe.As<Vector4, Vector128<float>>(ref this.V4R),
+               1);
+
+            Vector256<float> r5 = Avx.InsertVector128(
+               Unsafe.As<Vector4, Vector128<float>>(ref this.V1R).ToVector256(),
+               Unsafe.As<Vector4, Vector128<float>>(ref this.V5R),
+               1);
+
+            Vector256<float> r6 = Avx.InsertVector128(
+               Unsafe.As<Vector4, Vector128<float>>(ref this.V2R).ToVector256(),
+               Unsafe.As<Vector4, Vector128<float>>(ref this.V6R),
+               1);
+
+            Vector256<float> r7 = Avx.InsertVector128(
+               Unsafe.As<Vector4, Vector128<float>>(ref this.V3R).ToVector256(),
+               Unsafe.As<Vector4, Vector128<float>>(ref this.V7R),
+               1);
+
+            Vector256<float> t0 = Avx.UnpackLow(r0, r1);
+            Vector256<float> t2 = Avx.UnpackLow(r2, r3);
+            Vector256<float> v = Avx.Shuffle(t0, t2, 0x4E);
+            Unsafe.As<Vector4, Vector256<float>>(ref d.V0L) = Avx.Blend(t0, v, 0xCC);
+            Unsafe.As<Vector4, Vector256<float>>(ref d.V1L) = Avx.Blend(t2, v, 0x33);
+
+            Vector256<float> t4 = Avx.UnpackLow(r4, r5);
+            Vector256<float> t6 = Avx.UnpackLow(r6, r7);
+            v = Avx.Shuffle(t4, t6, 0x4E);
+            Unsafe.As<Vector4, Vector256<float>>(ref d.V4L) = Avx.Blend(t4, v, 0xCC);
+            Unsafe.As<Vector4, Vector256<float>>(ref d.V5L) = Avx.Blend(t6, v, 0x33);
+
+            Vector256<float> t1 = Avx.UnpackHigh(r0, r1);
+            Vector256<float> t3 = Avx.UnpackHigh(r2, r3);
+            v = Avx.Shuffle(t1, t3, 0x4E);
+            Unsafe.As<Vector4, Vector256<float>>(ref d.V2L) = Avx.Blend(t1, v, 0xCC);
+            Unsafe.As<Vector4, Vector256<float>>(ref d.V3L) = Avx.Blend(t3, v, 0x33);
+
+            Vector256<float> t5 = Avx.UnpackHigh(r4, r5);
+            Vector256<float> t7 = Avx.UnpackHigh(r6, r7);
+            v = Avx.Shuffle(t5, t7, 0x4E);
+            Unsafe.As<Vector4, Vector256<float>>(ref d.V6L) = Avx.Blend(t5, v, 0xCC);
```

Comment:
@tannergooding can we rely on codegen to produce optimal assembly when working with ` Unsafe.As`? (In comparison to `Store` / `LoadVector256` )

We wanted to avoid pinning, if possible.

Reply:
I would also like to subscribe to this newsletter.

Reply:
We should still fold the loads/stores. I believe there are a couple edge cases where we may generate an additional `lea` instruction, but the disassembly would nerd to be checked to see if it's an issue.

Reply:
Is there a new language feature I am unaware of that allows this without pinning?

https://source.dot.net/#System.Private.CoreLib/Matrix4x4.cs,273

```c#
Vector128<float> M11 = AdvSimd.LoadVector128(&value1.M11);
```

Reply:
In this case `Matrix4x4` is passed by value to the method, so the compiler can make sure `value1` is living on the stack, which means it never gets moved around by the GC. => the address can be used safely without pinning anything

Reply:
Actually: `JpegBlockPostprocessor` is the owner of all 3 blocks passed to `TransformIDCT`, and that struct always lives on the stack:
https://github.com/SixLabors/ImageSharp/blob/78a584e8482b052d7a9885682299e2f37518d83d/src/ImageSharp/Formats/Jpeg/Components/Decoder/JpegComponentPostProcessor.cs#L79

Therefore, you can get a pointer to the whole `Block8x8` buffer at the beginning of the method "safely" with Unsafe:
```C#
float* dPtr = (float*)Unsafe.AsPointer(ref d);
```

Reply:
Ah yes of course. This all makes sense thanks! 

I might give the pointer approach a try tomorrow just to see if there is any difference.

Reply:
Decided against it. I don't want to risk changes to external code breaking this.